dvc/repo/imp.py

```diff
@@ -1,13 +1,25 @@
+import os
+
+from dvc.utils.compat import urlparse, str, builtin_str
+
 from dvc.repo.scm_context import scm_context
+from dvc.output import Output
 
 
 @scm_context
-def imp(self, url, out, resume=False, fname=None):
+def imp(self, src, dst=None, resume=False, fname=None):
     from dvc.stage import Stage
 
+    if not dst:
+        if isinstance(src, str) or isinstance(src, builtin_str):
+            p = src
+        else:
+            p = src[Output.PARAM_PATH]
+        dst = os.path.basename(urlparse(p).path)
+
     with self.state:
         stage = Stage.create(
-            repo=self, cmd=None, deps=[url], outs=[out], fname=fname
+            repo=self, cmd=None, deps=[src], outs=[dst], fname=fname
```

Comment:
below: when is it possible that stage is None after this call? what does it mean for a user when the command silently returns None?

Reply:
ah, I see it's another problem that Stage.create take a part in the UX (with that logger info), ideally caller of the `imp` should translate `None` into some meaningful message.

Reply:
> below: when is it possible that stage is None after this call? what does it mean for a user when the command silently returns None?

It means that the stage is cached (like build cache) already.

Reply:
Yes, I got the idea. It's a problem with the way we do UX (with logger inside the code). Like I mentioned it should be done outside. It's not a suggestion for this PR for sure.